scripts/whatsdue.js

```diff
@@ -97,9 +97,14 @@ module.exports = function (robot) {
 
     /**
      * Robot responds to a message containing `!whatsdue`.
-     * Requires at least one course.
      */
-    robot.respond(/!?whatsdue ((?:[a-z]{4}[0-9]{4} ?)+)/i, function (res) {
+    robot.respond(/!?whatsdue ?((?:[a-z]{4}[0-9]{4} ?)+)?/i, function (res) {
+        // If there are no matching groups, throw an error.
+        if (!res.match[1]) {
```

Comment:
res.match[1] is only populated by valid matches, if it doesn't exist; there are no matches. Is there another check you have in mind/can recommend?